Darwin/Services/StitchRemoteMongoDBService/StitchSyncPerformanceTests/SyncL2ROnlyPerformanceTestDefinitions.swift

```diff
@@ -0,0 +1,251 @@
+
+import XCTest
+import Foundation
+import MongoSwift
+@testable import StitchRemoteMongoDBService
+@testable import StitchCoreRemoteMongoDBService
+
+class SyncL2ROnlyPerformanceTestDefinitions {
+    private static let TAG = SyncL2ROnlyPerformanceTestDefinitions.self
+
+    // NOTE: Many of the tests above 1024 bytes and above 500 docs will fail for various
+    // reasons because they hit undocumented limits. These failures along with stacktraces will
+    // be present in the reported results
+    private let docSizes = [1024, 2048, 5120, 10240, 25600, 51200, 102400]
+    private let numDocs = [100, 500, 1000, 5000, 10000, 25000]
+
+    static func testInitialSync(testHarness: SyncPerformanceIntTestHarness, runId: ObjectId) {
+        let testName = "testL2R_InitialSync"
+        print(testName)
+
+        // Local variable for list of documents captured by the test definition closures below.
+        // This should change for each iteration of the test.
+        var documentsForCurrentTest: [Document] = []
+
+        testHarness.runPerformanceTestWithParameters(
+            testName: testName,
+            runId: runId,
+            testDefinition: { (ctx, numDocs, docSize) in
+                // Initial sync for a purely L2R scenario means inserting local documents,
+                // and performing a sync pass to sync those documents up to the remote.
+                let sync = ctx.coll.sync
+
+                // If sync fails for any reason, halt the test
+                sync.configure(
+                    conflictHandler: DefaultConflictHandler<Document>.remoteWins(),
+                    changeEventDelegate: nil,
+                    errorListener: { error, id in
+                        print(
+                        "unexpected sync error with id " +
+                            "\(String(describing: id)): \(error)"
+                        )
+                        fatalError(error.localizedDescription)
+                    }
+                )
+
+                ctx.coll.sync.insertMany(&documentsForCurrentTest)
+
+                // Halt the test if the sync pass failed
+                guard let passed = try? ctx.coll.sync.proxy.dataSynchronizer.doSyncPass(),
+                    passed else {
+                    XCTFail("sync pass failed")
+                    return
+                }
+            }, beforeEach: { (ctx, numDocs, docSize) in
+                // Generate the documents that are to be synced via L2R
+                print("Setting up \(testName) for \(numDocs) \(docSize)-byte docs")
+                documentsForCurrentTest = SyncPerformanceTestUtils.generateDocuments(numDoc: numDocs, docSize: docSize)
+            }, afterEach: { (ctx, numDocs, _) in
+                // Verify that the test did indeed synchronize
+                // the provided documents remotely
+                let numOfDocsSynced = ctx.coll.count(Document())
+                if numDocs != numOfDocsSynced {
```

Comment:
In TestDefinitions I put a special throwing assertEqual() because we do not want to XCTFail() in this case, we just want to throw and let the test continue. Maybe we should move that method to utils though